pkg/sql/catalog/descpb/structured.proto

```diff
@@ -562,6 +562,13 @@ message IndexDescriptor {
   // TODO(mgartner): Update the comment to explain that columns are referenced
   // by their ID once #49766 is addressed.
   optional string predicate = 23 [(gogoproto.nullable) = false];
+
+  // UseDeletePreservingEncoding, if true, causes the index to be encoded with
+  // an additional bit that indicates whether or not the value has been deleted.
+  // This is necessary to preserve the delete history for the MVCC-compatible
+  // index backfiller
+  // `docs/RFCS/20211004_incremental_index_backfiller.md#new-index-encoding-for-deletions-vs-mvcc`
+  optional bool use_delete_preserving_encoding = 24 [(gogoproto.nullable) = false];
```

Comment:
Is there a reason why this field needs to be added, vs extending the `descpb.IndexDescriptorVersion` enum? We already have two fields in this message that specify how indexes are encoded, I'm quite reluctant to add a third unless there's a very good reason. This proto schema is messy enough as it is I don't want to make it worse.

Currently it seems to me that you could achieve the same results by adding:
- an enum value `PrimaryIndexWithDeletePreservingVersion` to supersede `PrimaryIndexWithStoredColumnsVersion`,
- an enum value  `SecondaryIndexWithDeletePreservingVersion` to supersede `StrictIndexColumnIDGuaranteesVersion`.

Reply:
In the RFC discussion we debated adding a new entry in the format enum vs a separate flag. I argued for new flag, since it seemed independent of the format -- whatever the format it, we can then box that and store it (or the fact it was deleted), so if there are 10 new formats, we don't need 10 new delete-preserving versions of them.

Reply:
That's a good-enough reason for me, thanks for clarifying.

Reply:
I feel this is worth mentioning in the field's comments. I know there's a link to the RFC but I'm guessing most people will be just as lazy as I was and won't go there.